src/OrchardCore/OrchardCore/Modules/Extensions/ServiceCollectionExtensions.cs

```diff
@@ -166,13 +168,21 @@ private static void AddAntiForgery(OrchardCoreBuilder builder)
             {
                 var settings = serviceProvider.GetRequiredService<ShellSettings>();
 
-                var tenantName = settings.Name;
+                var cookieName = "orchantiforgery_" + settings.Name;
```

Comment:
I assume that otherwise the name of the cookie in a Setup context is the same as the standard one in ASP.NET?
In this case can't we just delete two cookies? The one with the standard name, and the one for the tenant name? In the GET method of the setup?

Reply:
Here, when the tenant is not initialized, we 1st still use the host level middleware, so setting the cookie name at the tenant level has no effect, the host middleware will still use the host level options with the standard cookie name.

So, here we set the tenant cookie name just to retrieve it on tenant setup (e.g in place of using an hard coded string) and to be able to remove an eventual previous tenant level cookie.

Reply:
I am suggesting to delete well-known cookies, nothing to do with configuration.

For instance, what if in the controller we would delete the cookies name `"orchantiforgery_" + settings.Name` and any starting with `.AspNetCore.Antiforgery.`

Reply:
Ah okay, i will try it